Sources/Icon/IconTransaction.swift

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2017-2018 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import Foundation
+import BigInt
+
+public struct IconTransaction {
+    public let from: IconAddress
+    public let to: IconAddress
+    public let value: BigInt
+    public let fee: BigInt
+    public let timestamp: String
+    public let nonce: BigInt
+    public var tx_hash: Data
+    public var signature: Data
+
+    public init(
+        from: IconAddress,
+        to: IconAddress,
+        value: BigInt,
+        fee: BigInt,
+        timestamp: String,
+        nonce: BigInt,
+        tx_hash: Data = Data(),
+        signature: Data = Data()
+    ) {
+        self.from = from
+        self.to = to
+        self.value = value
+        self.fee = fee
+        self.timestamp = timestamp
+        self.nonce = nonce
+        self.tx_hash = tx_hash
+        self.signature = signature
+    }
+
+    public mutating func hash() {
```

Comment:
I think we don't need to store the hash value here, it's better to create a `IconSigner` to handle hash and signing

Reply:
The thing about storing the hash value is because it is used to broadcast the transaction to the ICO network.

Here is an example of the JSON API and all the data needed for broadcasting the transaction.
{
    "jsonrpc": "2.0",
    "method": "icx_sendTransaction",
    "id": 2,
    "params": {
        "from": "hxbe258ceb872e08851f1f59694dac2558708ece11",
        "to": "hxb0776ee37f5b45bfaea8cff1d8232fbb6122ec32",
        "value": "0xde0b6b3a7640000",
        "fee": "0x2386f26fc10000",
        "timestamp": "1516942975500598",
        "nonce": "8367273",
        "tx_hash": "4bf74e6aeeb43bde5dc8d5b62537a33ac8eb7605ebbdb51b015c1881b45b3aed",
        "signature": "VAia7YZ2Ji6igKWzjR2YsGa2m53nKPrfK7uXYW78QLE+ATehAVZPC40szvAiA6NEU5gCYB4c4qaQzqDh2ugcHgA="
    }
}
So I was thinking it's more convenient to have all the data needed in the same class for the transaction request.